scripts/o3de/tests/test_project_manager_interface.py

```diff
@@ -0,0 +1,413 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import pytest
+from unittest.mock import patch
+from inspect import signature
+import pathlib
+
+from o3de import manifest, engine_properties, register, cmake, engine_template, enable_gem, disable_gem, project_properties, repo, download
+
+# If any tests are failing in this, this means the interface Project Manager depends on has changed.
+# This likely means that some Project Manager functionality has been broken.
+# Please ensure that functionality found in o3de/Code/Tools/ProjectManager/Source/PythonBindings.cpp
+# still works with the affected interface before updating the test to the latest interface.
+
+
+# manifest interface
+def test_get_engine_json_data():
+    sig = signature(manifest.get_engine_json_data)
+    assert sig.parameters.__len__() >= 2
```

Comment:
nit: calling special attributes directly like this should typically be avoided, prefer to use the general purpose `len(...)` instead.

Reply:
Thanks for the reminder, haven't touched python in a bit :)